internal/pkg/cli/env_package.go

```diff
@@ -0,0 +1,298 @@
+	if o.envCfg != nil {
+		return o.envCfg, nil
+	}
+	cfg, err := o.cfgStore.GetEnvironment(o.appName, o.envName)
+	if err != nil {
+		return nil, fmt.Errorf("get environment %q in application %q: %w", o.envName, o.appName, err)
+	}
+	o.envCfg = cfg
+	return o.envCfg, nil
+}
+
+func (o *packageEnvOpts) validateOrAskEnvName() error {
+	if o.envName != "" {
+		if _, err := o.getEnvCfg(); err != nil {
+			log.Errorf("It seems like environment %s is not added in application %s yet. Have you run %s?\n",
+				o.envName, o.appName, color.HighlightCode("copilot env init"))
+			return err
+		}
+		return nil
+	}
+
+	name, err := o.sel.LocalEnvironment("Select an environment manifest from your workspace", "")
+	if err != nil {
+		return fmt.Errorf("select environment: %w", err)
+	}
+	o.envName = name
+	return nil
+}
+
+func (o *packageEnvOpts) environmentManifest() (*manifest.Environment, error) {
+	envCfg, err := o.getEnvCfg()
+	if err != nil {
+		return nil, err
+	}
+	raw, err := o.ws.ReadEnvironmentManifest(envCfg.Name)
+	if err != nil {
+		return nil, fmt.Errorf("read manifest for environment %q: %w", envCfg.Name, err)
+	}
+	interpolated, err := o.newInterpolator(o.appName, o.envName).Interpolate(string(raw))
+	if err != nil {
+		return nil, fmt.Errorf("interpolate environment variables for %q manifest: %w", o.envName, err)
+	}
+	mft, err := manifest.UnmarshalEnvironment([]byte(interpolated))
+	if err != nil {
+		return nil, fmt.Errorf("unmarshal environment manifest for %q: %w", envCfg.Name, err)
+	}
+	if err := mft.Validate(); err != nil {
+		return nil, fmt.Errorf("validate environment manifest for %q: %w", envCfg.Name, err)
+	}
+	return mft, nil
+}
+
+func (o *packageEnvOpts) setWriters() error {
+	if o.outputDir == "" {
+		return nil
+	}
+	if err := o.fs.MkdirAll(o.outputDir, 0755); err != nil {
+		return fmt.Errorf("create directory %q: %w", o.outputDir, err)
+	}
+
+	path := filepath.Join(o.outputDir, fmt.Sprintf("env-%s.yml", o.envName))
+	tplFile, err := o.fs.Create(path)
+	if err != nil {
+		return fmt.Errorf("create file at %q: %w", path, err)
+	}
+	path = filepath.Join(o.outputDir, fmt.Sprintf("env-%s.params.json", o.envName))
+	paramsFile, err := o.fs.Create(path)
+	if err != nil {
+		return fmt.Errorf("create file at %q: %w", path, err)
+	}
+
+	o.tplWriter = tplFile
+	o.paramsWriter = paramsFile
+	return nil
+}
+
+func (o *packageEnvOpts) write(wc io.WriteCloser, dat string) error {
+	if _, err := wc.Write([]byte(dat)); err != nil {
+		return err
+	}
+	return wc.Close()
+}
+
+// buildEnvPkgCmd builds the command for printing an environment CloudFormation stack configuration.
+func buildEnvPkgCmd() *cobra.Command {
+	vars := packageEnvVars{}
+	cmd := &cobra.Command{
+		Use:   "package",
+		Short: "Prints the AWS CloudFormation template of an environment.",
+		Long:  `Prints the CloudFormation stack template and configuration used to deploy an environment.`,
+		Example: `
+  Print the CloudFormation template for the "prod" environment.
```

Comment:
nit: should we be consistent about "print" or "prints"? Maybe
```suggestion
		Short: "Print the AWS CloudFormation template of an environment.",
		Long:  `Print the CloudFormation stack template and configuration used to deploy an environment.`,
		Example: `
  Print the CloudFormation template for the "prod" environment.
```

Reply:
Done!